File: adt2sys/BlockReadF.c
// This file is part of Adlib Tracker II (AT2).
//
// SPDX-FileType: SOURCE
// SPDX-FileCopyrightText: 2014-2025 The Adlib Tracker 2 Authors
// SPDX-License-Identifier: GPL-3.0-or-later

void BlockReadF (void *file, void *data, int32_t size, int32_t *bytes_read)
{
  DBG_ENTER ("BlockReadF");

  Pascal_BlockRead (file, data, size, bytes_read);
  if (Pascal_IOResult () != 0) *bytes_read = 0;

  DBG_LEAVE (); //EXIT //BlockReadF
}

File: adt2sys/pas/BlockReadF.pas
// This file is part of Adlib Tracker II (AT2).
//
// SPDX-FileType: SOURCE
// SPDX-FileCopyrightText: 2014-2025 The Adlib Tracker 2 Authors
// SPDX-License-Identifier: GPL-3.0-or-later

procedure BlockReadF(var f: File; var data; size: Longint; var bytes_read: Longint);
begin
  _dbg_enter ({$I %FILE%}, 'BlockReadF');

  {$i-}
  BlockRead(f,data,size,bytes_read);
  {$i+}
  If (IOresult <> 0) then
    bytes_read := 0;

  _dbg_leave; //EXIT //BlockReadF
end;
